-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0d1b43/51951/summary.html
**COMMIT**: 3a28fefeb522cad47eba392c6c9e42d9e3fe2601
**CMSSW**: CMSSW_16_1_EVOLUTION_X_2026-03-09-2300/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/50379/51951/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @makortel cms-sw/cmssw#50332
 - @ghyls cms-sw/cmssw#50335
 - @JanGerritSchulz cms-sw/cmssw#50318
 - @bsunanda cms-sw/cmssw#50355

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0d1b43/51951/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0d1b43/51951/git-merge-result


## Failed Build

I found compilation error when building: 

<pre>>> Compiling edm plugin src/IOMC/ParticleGuns/src/RandomXiThetaGunProducer.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_16_1_EVOLUTION_X_2026-03-09-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_16_1_EVOLUTION_X_2026-03-09-2300' -Isrc -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/pcre/8.43-6d98fda3bfd074ebb583e2d6a2c75d25/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/bz2lib/1.0.6-d113e1c6278c07eeaff5f84db9548446/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/clhep/2.4.7.2-8ba7d0a07f7f2205ab0e6cfc6c414f34/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/gsl/2.6-9011a41928244b609ca4c22c439b3fef/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/hepmc/2.06.10-903e23142392c852f3df97c32fbd4659/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/heppdt/3.04.01-52cfb8b2a7df91accb25807698381dba/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/libuuid/2.34-5ba7a8abfc0c5fecdc448cca360c25ff/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/lcg/root/6.36.09-8a6db48e639169d76071b7f4a20dbfbf/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/tbb/v2022.3.0-88eb7be4ee320d604a798a914aea6359/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/xz/5.6.4-b9c4ffbc390ed320a5d57fd552e29a05/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/zlib/1.2.13-589f6bb51bbeba38a7adf5a10ea8a093/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/hepmc3/3.2.7-f8f0da18d94695d72ff740910e834c9f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/OpenBLAS/0.3.27-da4a3c2bb8ae43f3913a4a44acdb1b50/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/tinyxml2/6.2.0-67924ead96ecb4e69aad321b767979a5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/IOMC/ParticleGuns/src/IOMCParticleGuns/RandomXiThetaGunProducer.cc.d src/IOMC/ParticleGuns/src/RandomXiThetaGunProducer.cc -o tmp/el8_amd64_gcc13/src/IOMC/ParticleGuns/src/IOMCParticleGuns/RandomXiThetaGunProducer.cc.o
>> Compiling edm plugin src/IOMC/ParticleGuns/src/SealModule.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_16_1_EVOLUTION_X_2026-03-09-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_16_1_EVOLUTION_X_2026-03-09-2300' -Isrc -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/pcre/8.43-6d98fda3bfd074ebb583e2d6a2c75d25/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/bz2lib/1.0.6-d113e1c6278c07eeaff5f84db9548446/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/clhep/2.4.7.2-8ba7d0a07f7f2205ab0e6cfc6c414f34/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/gsl/2.6-9011a41928244b609ca4c22c439b3fef/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/hepmc/2.06.10-903e23142392c852f3df97c32fbd4659/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/heppdt/3.04.01-52cfb8b2a7df91accb25807698381dba/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/libuuid/2.34-5ba7a8abfc0c5fecdc448cca360c25ff/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/lcg/root/6.36.09-8a6db48e639169d76071b7f4a20dbfbf/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/tbb/v2022.3.0-88eb7be4ee320d604a798a914aea6359/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/xz/5.6.4-b9c4ffbc390ed320a5d57fd552e29a05/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/zlib/1.2.13-589f6bb51bbeba38a7adf5a10ea8a093/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/hepmc3/3.2.7-f8f0da18d94695d72ff740910e834c9f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/OpenBLAS/0.3.27-da4a3c2bb8ae43f3913a4a44acdb1b50/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02932/el8_amd64_gcc13/external/tinyxml2/6.2.0-67924ead96ecb4e69aad321b767979a5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/IOMC/ParticleGuns/src/IOMCParticleGuns/SealModule.cc.d src/IOMC/ParticleGuns/src/SealModule.cc -o tmp/el8_amd64_gcc13/src/IOMC/ParticleGuns/src/IOMCParticleGuns/SealModule.cc.o
In file included from src/IOMC/ParticleGuns/src/HTo2XTo4LGunProducer.cc:5:
src/SimDataFormats/GeneratorProducts/interface/HepMCProduct.h:58:9: error: conflicting declaration 'using edm::HepMCProduct = class edm::io_v1::HepMCProduct'
   58 |   using HepMCProduct = io_v1::HepMCProduct;
      |         ^~~~~~~~~~~~
In file included from src/IOMC/ParticleGuns/src/HTo2XTo4LGunProducer.cc:3:
src/IOMC/ParticleGuns/interface/HTo2XTo4LGunProducer.h:29:9: note: previous declaration as 'class edm::HepMCProduct'
   29 |   class HepMCProduct;
</pre>

--- a/DataFormats/FTLRecHit/interface/FTLCluster.h
+++ b/DataFormats/FTLRecHit/interface/FTLCluster.h
@@ -17,276 +17,278 @@
 
 #include "DataFormats/DetId/interface/DetId.h"
 
-class FTLCluster {
-public:
-  typedef DetId key_type;
-
-  class FTLHit {
+namespace io_v1 {
+  class FTLCluster {
   public:
-    constexpr FTLHit() : x_(0), y_(0), energy_(0), time_(0), time_error_(0) {}
-    constexpr FTLHit(uint16_t hit_x, uint16_t hit_y, float hit_energy, float hit_time, float hit_time_error)
-        : x_(hit_x), y_(hit_y), energy_(hit_energy), time_(hit_time), time_error_(hit_time_error) {}
-    constexpr uint16_t x() { return x_; }
-    constexpr uint16_t y() { return y_; }
-    constexpr uint16_t energy() { return energy_; }
-    constexpr uint16_t time() { return time_; }
-    constexpr uint16_t time_error() { return time_error_; }
+    typedef DetId key_type;
 
-  private:
-    uint16_t x_;  //row
-    uint16_t y_;  //col
-    float energy_;
-    float time_;
-    float time_error_;
-  };
+    class FTLHit {
+    public:
+      constexpr FTLHit() : x_(0), y_(0), energy_(0), time_(0), time_error_(0) {}
+      constexpr FTLHit(uint16_t hit_x, uint16_t hit_y, float hit_energy, float hit_time, float hit_time_error)
+          : x_(hit_x), y_(hit_y), energy_(hit_energy), time_(hit_time), time_error_(hit_time_error) {}
+      constexpr uint16_t x() { return x_; }
+      constexpr uint16_t y() { return y_; }
+      constexpr uint16_t energy() { return energy_; }
+      constexpr uint16_t time() { return time_; }
+      constexpr uint16_t time_error() { return time_error_; }
 
-  //--- Integer shift in x and y directions.
-  class Shift {
-  public:
-    constexpr Shift(int dx, int dy) : dx_(dx), dy_(dy) {}
-    constexpr Shift() : dx_(0), dy_(0) {}
-    constexpr int dx() const { return dx_; }
-    constexpr int dy() const { return dy_; }
+    private:
+      uint16_t x_;  //row
+      uint16_t y_;  //col
+      float energy_;
+      float time_;
+      float time_error_;
+    };
 
-  private:
-    int dx_;
-    int dy_;
-  };
+    //--- Integer shift in x and y directions.
+    class Shift {
+    public:
+      constexpr Shift(int dx, int dy) : dx_(dx), dy_(dy) {}
+      constexpr Shift() : dx_(0), dy_(0) {}
+      constexpr int dx() const { return dx_; }
+      constexpr int dy() const { return dy_; }
 
-  //--- Position of a FTL Hit
-  class FTLHitPos {
-  public:
-    constexpr FTLHitPos() : row_(0), col_(0) {}
-    constexpr FTLHitPos(int row, int col) : row_(row), col_(col) {}
-    constexpr int row() const { return row_; }
-    constexpr int col() const { return col_; }
-    constexpr FTLHitPos operator+(const Shift& shift) const {
-      return FTLHitPos(row() + shift.dx(), col() + shift.dy());
-    }
+    private:
+      int dx_;
+      int dy_;
+    };
 
-  private:
-    int row_;
-    int col_;
-  };
+    //--- Position of a FTL Hit
+    class FTLHitPos {
+    public:
+      constexpr FTLHitPos() : row_(0), col_(0) {}
+      constexpr FTLHitPos(int row, int col) : row_(row), col_(col) {}
+      constexpr int row() const { return row_; }
+      constexpr int col() const { return col_; }
+      constexpr FTLHitPos operator+(const Shift& shift) const {
+        return FTLHitPos(row() + shift.dx(), col() + shift.dy());
+      }
 
-  static constexpr unsigned int MAXSPAN = 255;
-  static constexpr unsigned int MAXPOS = 2047;
+    private:
+      int row_;
+      int col_;
+    };
 
-  /** Construct from a range of digis that form a cluster and from 
+    static constexpr unsigned int MAXSPAN = 255;
+    static constexpr unsigned int MAXPOS = 2047;
+
+    /** Construct from a range of digis that form a cluster and from 
    *  a DetID. The range is assumed to be non-empty.
    */
-  FTLCluster() {}
+    FTLCluster() {}
 
-  FTLCluster(DetId id,
-             unsigned int isize,
-             float const* energys,
-             float const* times,
-             float const* time_errors,
-             uint16_t const* xpos,
-             uint16_t const* ypos,
-             uint16_t const xmin,
-             uint16_t const ymin)
-      : theid(id),
-        theHitOffset(2 * isize),
-        theHitENERGY(energys, energys + isize),
-        theHitTIME(times, times + isize),
-        theHitTIME_ERROR(time_errors, time_errors + isize) {
-    uint16_t maxCol = 0;
-    uint16_t maxRow = 0;
-    int maxHit = -1;
-    float maxEnergy = -99999;
-    for (unsigned int i = 0; i != isize; ++i) {
-      uint16_t xoffset = xpos[i] - xmin;
-      uint16_t yoffset = ypos[i] - ymin;
-      theHitOffset[i * 2] = std::min(uint16_t(MAXSPAN), xoffset);
-      theHitOffset[i * 2 + 1] = std::min(uint16_t(MAXSPAN), yoffset);
-      if (xoffset > maxRow)
-        maxRow = xoffset;
-      if (yoffset > maxCol)
-        maxCol = yoffset;
-      if (theHitENERGY[i] > maxEnergy) {
-        maxHit = i;
-        maxEnergy = theHitENERGY[i];
+    FTLCluster(DetId id,
+               unsigned int isize,
+               float const* energys,
+               float const* times,
+               float const* time_errors,
+               uint16_t const* xpos,
+               uint16_t const* ypos,
+               uint16_t const xmin,
+               uint16_t const ymin)
+        : theid(id),
+          theHitOffset(2 * isize),
+          theHitENERGY(energys, energys + isize),
+          theHitTIME(times, times + isize),
+          theHitTIME_ERROR(time_errors, time_errors + isize) {
+      uint16_t maxCol = 0;
+      uint16_t maxRow = 0;
+      int maxHit = -1;
+      float maxEnergy = -99999;
+      for (unsigned int i = 0; i != isize; ++i) {
+        uint16_t xoffset = xpos[i] - xmin;
+        uint16_t yoffset = ypos[i] - ymin;
+        theHitOffset[i * 2] = std::min(uint16_t(MAXSPAN), xoffset);
+        theHitOffset[i * 2 + 1] = std::min(uint16_t(MAXSPAN), yoffset);
+        if (xoffset > maxRow)
+          maxRow = xoffset;
+        if (yoffset > maxCol)
+          maxCol = yoffset;
+        if (theHitENERGY[i] > maxEnergy) {
+          maxHit = i;
+          maxEnergy = theHitENERGY[i];
+        }
+      }
+      packRow(xmin, maxRow);
+      packCol(ymin, maxCol);
+
+      if (maxHit >= 0)
+        seed_ = std::min(uint8_t(MAXSPAN), uint8_t(maxHit));
+    }
+
+    // linear average position (barycenter)
+    inline float x() const {
+      auto x_pos = [this](unsigned int i) { return this->theHitOffset[i * 2] + minHitRow(); };
+      return weighted_mean(this->theHitENERGY, x_pos);
+    }
+
+    inline float y() const {
+      auto y_pos = [this](unsigned int i) { return this->theHitOffset[i * 2 + 1] + minHitCol(); };
+      return weighted_mean(this->theHitENERGY, y_pos);
+    }
+
+    inline float positionError(const float sigmaPos) const {
+      float sumW2(0.f), sumW(0.f);
+      for (const auto& hitW : theHitENERGY) {
+        sumW2 += hitW * hitW;
+        sumW += hitW;
+      }
+      if (sumW > 0)
+        return sigmaPos * std::sqrt(sumW2) / sumW;
+      else
+        return -999.f;
+    }
+
+    inline float time() const {
+      auto t = [this](unsigned int i) { return this->theHitTIME[i]; };
+      return weighted_mean(this->theHitENERGY, t);
+    }
+
+    inline float timeError() const {
+      auto t_err = [this](unsigned int i) { return this->theHitTIME_ERROR[i]; };
+      return weighted_mean_error(this->theHitENERGY, t_err);
+    }
+
+    // Return number of hits.
+    inline int size() const { return theHitENERGY.size(); }
+
+    // Return cluster dimension in the x direction.
+    inline int sizeX() const { return rowSpan() + 1; }
+
+    // Return cluster dimension in the y direction.
+    inline int sizeY() const { return colSpan() + 1; }
+
+    inline float energy() const {
+      return std::accumulate(theHitENERGY.begin(), theHitENERGY.end(), 0.f);
+    }  // Return total cluster energy.
+
+    inline int minHitRow() const { return theMinHitRow; }             // The min x index.
+    inline int maxHitRow() const { return minHitRow() + rowSpan(); }  // The max x index.
+    inline int minHitCol() const { return theMinHitCol; }             // The min y index.
+    inline int maxHitCol() const { return minHitCol() + colSpan(); }  // The max y index.
+
+    const std::vector<uint8_t>& hitOffset() const { return theHitOffset; }
+    const std::vector<float>& hitENERGY() const { return theHitENERGY; }
+    const std::vector<float>& hitTIME() const { return theHitTIME; }
+    const std::vector<float>& hitTIME_ERROR() const { return theHitTIME_ERROR; }
+
+    // infinite faster than above...
+    FTLHit hit(int i) const {
+      return FTLHit(minHitRow() + theHitOffset[i * 2],
+                    minHitCol() + theHitOffset[i * 2 + 1],
+                    theHitENERGY[i],
+                    theHitTIME[i],
+                    theHitTIME_ERROR[i]);
+    }
+
+    FTLHit seed() const { return hit(seed_); }
+
+    int colSpan() const { return theHitColSpan; }
+
+    int rowSpan() const { return theHitRowSpan; }
+
+    const DetId& id() const { return theid; }
+    const DetId& detid() const { return id(); }
+
+    bool overflowCol() const { return overflow_(theHitColSpan); }
+
+    bool overflowRow() const { return overflow_(theHitRowSpan); }
+
+    bool overflow() const { return overflowCol() || overflowRow(); }
+
+    void packCol(uint16_t ymin, uint16_t yspan) {
+      theMinHitCol = ymin;
+      theHitColSpan = std::min(yspan, uint16_t(MAXSPAN));
+    }
+    void packRow(uint16_t xmin, uint16_t xspan) {
+      theMinHitRow = xmin;
+      theHitRowSpan = std::min(xspan, uint16_t(MAXSPAN));
+    }
+
+    void setClusterPosX(float posx) { pos_x = posx; }
+    void setClusterErrorX(float errx) { err_x = errx; }
+    void setClusterErrorTime(float errtime) { err_time = errtime; }
+    float getClusterPosX() const { return pos_x; }
+    float getClusterErrorX() const { return err_x; }
+    float getClusterErrorTime() const { return err_time; }
+
+  private:
+    DetId theid;
+
+    std::vector<uint8_t> theHitOffset;
+    std::vector<float> theHitENERGY;
+    std::vector<float> theHitTIME;
+    std::vector<float> theHitTIME_ERROR;
+
+    uint16_t theMinHitRow = MAXPOS;  // Minimum hit index in the x direction (low edge).
+    uint16_t theMinHitCol = MAXPOS;  // Minimum hit index in the y direction (left edge).
+    uint8_t theHitRowSpan = 0;       // Span hit index in the x direction (low edge).
+    uint8_t theHitColSpan = 0;       // Span hit index in the y direction (left edge).
+
+    float pos_x = -99999.9f;  // For pixels with internal position information in one coordinate (i.e. BTL crystals)
+    float err_x = -99999.9f;  // For pixels with internal position information in one coordinate (i.e. BTL crystals)
+    float err_time = -99999.9f;
+
+    uint8_t seed_;
+
+    template <typename SumFunc, typename OutFunc>
+    float weighted_sum(const std::vector<float>& weights, SumFunc&& sumFunc, OutFunc&& outFunc) const {
+      float tot = 0;
+      float sumW = 0;
+      for (unsigned int i = 0; i < weights.size(); ++i) {
+        tot += sumFunc(i);
+        sumW += weights[i];
+      }
+      return outFunc(tot, sumW);
+    }
+
+    template <typename Value>
+    float weighted_mean(const std::vector<float>& weights, Value&& value) const {
+      auto sumFunc = [&weights, value](unsigned int i) { return weights[i] * value(i); };
+      auto outFunc = [](float x, float y) {
+        if (y > 0)
+          return (float)x / y;
+        else
+          return -999.f;
+      };
+      return weighted_sum(weights, sumFunc, outFunc);
+    }
+
+    template <typename Err>
+    float weighted_mean_error(const std::vector<float>& weights, Err&& err) const {
+      auto sumFunc = [&weights, err](unsigned int i) { return weights[i] * weights[i] * err(i) * err(i); };
+      auto outFunc = [](float x, float y) {
+        if (y > 0)
+          return (float)sqrt(x) / y;
+        else
+          return -999.f;
+      };
+      return weighted_sum(weights, sumFunc, outFunc);
+    }
+
+    static int overflow_(uint16_t span) { return span == uint16_t(MAXSPAN); }
+  };
+
+  // Comparison operators  (needed by DetSetVector & SortedCollection )
+  inline bool operator<(const FTLCluster& one, const FTLCluster& other) {
+    if (one.detid() == other.detid()) {
+      if (one.minHitRow() < other.minHitRow()) {
+        return true;
+      } else if (one.minHitRow() > other.minHitRow()) {
+        return false;
+      } else if (one.minHitCol() < other.minHitCol()) {
+        return true;
+      } else {
+        return false;
       }
     }
-    packRow(xmin, maxRow);
-    packCol(ymin, maxCol);
-
-    if (maxHit >= 0)
-      seed_ = std::min(uint8_t(MAXSPAN), uint8_t(maxHit));
+    return one.detid() < other.detid();
   }
 
-  // linear average position (barycenter)
-  inline float x() const {
-    auto x_pos = [this](unsigned int i) { return this->theHitOffset[i * 2] + minHitRow(); };
-    return weighted_mean(this->theHitENERGY, x_pos);
-  }
+  inline bool operator<(const FTLCluster& one, const uint32_t& detid) { return one.detid() < detid; }
 
-  inline float y() const {
-    auto y_pos = [this](unsigned int i) { return this->theHitOffset[i * 2 + 1] + minHitCol(); };
-    return weighted_mean(this->theHitENERGY, y_pos);
-  }
-
-  inline float positionError(const float sigmaPos) const {
-    float sumW2(0.f), sumW(0.f);
-    for (const auto& hitW : theHitENERGY) {
-      sumW2 += hitW * hitW;
-      sumW += hitW;
-    }
-    if (sumW > 0)
-      return sigmaPos * std::sqrt(sumW2) / sumW;
-    else
-      return -999.f;
-  }
-
-  inline float time() const {
-    auto t = [this](unsigned int i) { return this->theHitTIME[i]; };
-    return weighted_mean(this->theHitENERGY, t);
-  }
-
-  inline float timeError() const {
-    auto t_err = [this](unsigned int i) { return this->theHitTIME_ERROR[i]; };
-    return weighted_mean_error(this->theHitENERGY, t_err);
-  }
-
-  // Return number of hits.
-  inline int size() const { return theHitENERGY.size(); }
-
-  // Return cluster dimension in the x direction.
-  inline int sizeX() const { return rowSpan() + 1; }
-
-  // Return cluster dimension in the y direction.
-  inline int sizeY() const { return colSpan() + 1; }
-
-  inline float energy() const {
-    return std::accumulate(theHitENERGY.begin(), theHitENERGY.end(), 0.f);
-  }  // Return total cluster energy.
-
-  inline int minHitRow() const { return theMinHitRow; }             // The min x index.
-  inline int maxHitRow() const { return minHitRow() + rowSpan(); }  // The max x index.
-  inline int minHitCol() const { return theMinHitCol; }             // The min y index.
-  inline int maxHitCol() const { return minHitCol() + colSpan(); }  // The max y index.
-
-  const std::vector<uint8_t>& hitOffset() const { return theHitOffset; }
-  const std::vector<float>& hitENERGY() const { return theHitENERGY; }
-  const std::vector<float>& hitTIME() const { return theHitTIME; }
-  const std::vector<float>& hitTIME_ERROR() const { return theHitTIME_ERROR; }
-
-  // infinite faster than above...
-  FTLHit hit(int i) const {
-    return FTLHit(minHitRow() + theHitOffset[i * 2],
-                  minHitCol() + theHitOffset[i * 2 + 1],
-                  theHitENERGY[i],
-                  theHitTIME[i],
-                  theHitTIME_ERROR[i]);
-  }
-
-  FTLHit seed() const { return hit(seed_); }
-
-  int colSpan() const { return theHitColSpan; }
-
-  int rowSpan() const { return theHitRowSpan; }
-
-  const DetId& id() const { return theid; }
-  const DetId& detid() const { return id(); }
-
-  bool overflowCol() const { return overflow_(theHitColSpan); }
-
-  bool overflowRow() const { return overflow_(theHitRowSpan); }
-
-  bool overflow() const { return overflowCol() || overflowRow(); }
-
-  void packCol(uint16_t ymin, uint16_t yspan) {
-    theMinHitCol = ymin;
-    theHitColSpan = std::min(yspan, uint16_t(MAXSPAN));
-  }
-  void packRow(uint16_t xmin, uint16_t xspan) {
-    theMinHitRow = xmin;
-    theHitRowSpan = std::min(xspan, uint16_t(MAXSPAN));
-  }
-
-  void setClusterPosX(float posx) { pos_x = posx; }
-  void setClusterErrorX(float errx) { err_x = errx; }
-  void setClusterErrorTime(float errtime) { err_time = errtime; }
-  float getClusterPosX() const { return pos_x; }
-  float getClusterErrorX() const { return err_x; }
-  float getClusterErrorTime() const { return err_time; }
-
-private:
-  DetId theid;
-
-  std::vector<uint8_t> theHitOffset;
-  std::vector<float> theHitENERGY;
-  std::vector<float> theHitTIME;
-  std::vector<float> theHitTIME_ERROR;
-
-  uint16_t theMinHitRow = MAXPOS;  // Minimum hit index in the x direction (low edge).
-  uint16_t theMinHitCol = MAXPOS;  // Minimum hit index in the y direction (left edge).
-  uint8_t theHitRowSpan = 0;       // Span hit index in the x direction (low edge).
-  uint8_t theHitColSpan = 0;       // Span hit index in the y direction (left edge).
-
-  float pos_x = -99999.9f;  // For pixels with internal position information in one coordinate (i.e. BTL crystals)
-  float err_x = -99999.9f;  // For pixels with internal position information in one coordinate (i.e. BTL crystals)
-  float err_time = -99999.9f;
-
-  uint8_t seed_;
-
-  template <typename SumFunc, typename OutFunc>
-  float weighted_sum(const std::vector<float>& weights, SumFunc&& sumFunc, OutFunc&& outFunc) const {
-    float tot = 0;
-    float sumW = 0;
-    for (unsigned int i = 0; i < weights.size(); ++i) {
-      tot += sumFunc(i);
-      sumW += weights[i];
-    }
-    return outFunc(tot, sumW);
-  }
-
-  template <typename Value>
-  float weighted_mean(const std::vector<float>& weights, Value&& value) const {
-    auto sumFunc = [&weights, value](unsigned int i) { return weights[i] * value(i); };
-    auto outFunc = [](float x, float y) {
-      if (y > 0)
-        return (float)x / y;
-      else
-        return -999.f;
-    };
-    return weighted_sum(weights, sumFunc, outFunc);
-  }
-
-  template <typename Err>
-  float weighted_mean_error(const std::vector<float>& weights, Err&& err) const {
-    auto sumFunc = [&weights, err](unsigned int i) { return weights[i] * weights[i] * err(i) * err(i); };
-    auto outFunc = [](float x, float y) {
-      if (y > 0)
-        return (float)sqrt(x) / y;
-      else
-        return -999.f;
-    };
-    return weighted_sum(weights, sumFunc, outFunc);
-  }
-
-  static int overflow_(uint16_t span) { return span == uint16_t(MAXSPAN); }
-};
-
-// Comparison operators  (needed by DetSetVector & SortedCollection )
-inline bool operator<(const FTLCluster& one, const FTLCluster& other) {
-  if (one.detid() == other.detid()) {
-    if (one.minHitRow() < other.minHitRow()) {
-      return true;
-    } else if (one.minHitRow() > other.minHitRow()) {
-      return false;
-    } else if (one.minHitCol() < other.minHitCol()) {
-      return true;
-    } else {
-      return false;
-    }
-  }
-  return one.detid() < other.detid();
-}
-
-inline bool operator<(const FTLCluster& one, const uint32_t& detid) { return one.detid() < detid; }
-
-inline bool operator<(const uint32_t& detid, const FTLCluster& other) { return detid < other.detid(); }
-
+  inline bool operator<(const uint32_t& detid, const FTLCluster& other) { return detid < other.detid(); }
+}  // namespace io_v1
+using FTLCluster = io_v1::FTLCluster;
 #endif

--- a/DataFormats/FTLRecHit/interface/FTLRecHit.h
+++ b/DataFormats/FTLRecHit/interface/FTLRecHit.h
@@ -12,18 +12,19 @@
  * \author Lindsey Gray
  */
 
-class FTLRecHit {
-public:
-  typedef DetId key_type;
+namespace io_v1 {
+  class FTLRecHit {
+  public:
+    typedef DetId key_type;
 
-  // FTLEE recHit flags
-  enum Flags {
-    kGood = 0,  // channel ok, the energy and time measurement are reliable
-    kKilled,    // MC only flag: the channel is killed in the real detector
-    kUnknown    // to ease the interface with functions returning flags.
-  };
+    // FTLEE recHit flags
+    enum Flags {
+      kGood = 0,  // channel ok, the energy and time measurement are reliable
+      kKilled,    // MC only flag: the channel is killed in the real detector
+      kUnknown    // to ease the interface with functions returning flags.
+    };
 
-  /** bit structure of CaloRecHit::flags_ used in FTLRecHit:
+    /** bit structure of CaloRecHit::flags_ used in FTLRecHit:
    *
    *  | 32 | 31...25 | 24...12 | 11...5 | 4...1 |
    *     |      |         |         |       |
@@ -34,79 +35,80 @@ public:
    *     +--> spare                                               ( 1 bit )
    */
 
-  FTLRecHit();
-  // by default a recHit is greated with no flag
-  FTLRecHit(const DetId& id,
-            uint8_t row,
-            uint8_t column,
-            float energy,
-            float time,
-            float timeError,
-            float position,
-            float positionError,
-            uint32_t flagBits = 0);
+    FTLRecHit();
+    // by default a recHit is greated with no flag
+    FTLRecHit(const DetId& id,
+              uint8_t row,
+              uint8_t column,
+              float energy,
+              float time,
+              float timeError,
+              float position,
+              float positionError,
+              uint32_t flagBits = 0);
 
-  FTLRecHit(const DetId& id,
-            float energy,
-            float time,
-            float timeError,
-            float position,
-            float positionError,
-            uint32_t flagBits = 0);
+    FTLRecHit(const DetId& id,
+              float energy,
+              float time,
+              float timeError,
+              float position,
+              float positionError,
+              uint32_t flagBits = 0);
 
-  /// get the id
+    /// get the id
 
-  float energy() const { return energy_; }
-  void setEnergy(float energy) { energy_ = energy; }
+    float energy() const { return energy_; }
+    void setEnergy(float energy) { energy_ = energy; }
 
-  const DetId& id() const { return id_; }
-  const DetId& detid() const { return id(); }
+    const DetId& id() const { return id_; }
+    const DetId& detid() const { return id(); }
 
-  const MTDDetId mtdId() const { return MTDDetId(id_); }
+    const MTDDetId mtdId() const { return MTDDetId(id_); }
 
-  int row() const { return row_; }
-  int column() const { return column_; }
+    int row() const { return row_; }
+    int column() const { return column_; }
 
-  float time() const { return time_; }
-  void setTime(float time) { time_ = time; }
+    float time() const { return time_; }
+    void setTime(float time) { time_ = time; }
 
-  float position() const { return position_; }
-  void setPosition(float position) { position_ = position; }
+    float position() const { return position_; }
+    void setPosition(float position) { position_ = position; }
 
-  bool isTimeValid() const;
-  bool isTimeErrorValid() const;
+    bool isTimeValid() const;
+    bool isTimeErrorValid() const;
 
-  float timeError() const { return timeError_; }
-  void setTimeError(float err) { timeError_ = err; }
+    float timeError() const { return timeError_; }
+    void setTimeError(float err) { timeError_ = err; }
 
-  float positionError() const { return positionError_; }
-  void setPositionError(float poserr) { positionError_ = poserr; }
+    float positionError() const { return positionError_; }
+    void setPositionError(float poserr) { positionError_ = poserr; }
 
-  /// set the flags (from Flags or ESFlags)
-  void setFlag(int flag) { flagBits_ |= (0x1 << flag); }
-  void unsetFlag(int flag) { flagBits_ &= ~(0x1 << flag); }
+    /// set the flags (from Flags or ESFlags)
+    void setFlag(int flag) { flagBits_ |= (0x1 << flag); }
+    void unsetFlag(int flag) { flagBits_ &= ~(0x1 << flag); }
 
-  /// check if the flag is true
-  bool checkFlag(int flag) const { return flagBits_ & (0x1 << flag); }
+    /// check if the flag is true
+    bool checkFlag(int flag) const { return flagBits_ & (0x1 << flag); }
 
-  /// check if one of the flags in a set is true
-  bool checkFlags(const std::vector<int>& flagsvec) const;
+    /// check if one of the flags in a set is true
+    bool checkFlags(const std::vector<int>& flagsvec) const;
 
-private:
-  DetId id_;
-  float energy_;
-  float time_;
-  float timeError_;
-  /// position is the distance from the center of the bar to hit
-  float position_;
-  float positionError_;
-  uint8_t row_;
-  uint8_t column_;
+  private:
+    DetId id_;
+    float energy_;
+    float time_;
+    float timeError_;
+    /// position is the distance from the center of the bar to hit
+    float position_;
+    float positionError_;
+    uint8_t row_;
+    uint8_t column_;
 
-  /// store rechit condition (see Flags enum) in a bit-wise way
-  unsigned char flagBits_;
-};
+    /// store rechit condition (see Flags enum) in a bit-wise way
+    unsigned char flagBits_;
+  };
 
-std::ostream& operator<<(std::ostream& s, const FTLRecHit& hit);
-
+  std::ostream& operator<<(std::ostream& s, const FTLRecHit& hit);
+}  // namespace io_v1
+using FTLRecHit = io_v1::FTLRecHit;
 #endif

--- a/DataFormats/FTLRecHit/interface/FTLRecHitComparison.h
+++ b/DataFormats/FTLRecHit/interface/FTLRecHitComparison.h
@@ -5,15 +5,16 @@
 
 //ordering capability mandatory for lazy getter framework
 // Comparison operators
-inline bool operator<(const FTLRecHit& one, const FTLRecHit& other) {
-  if (one.detid() == other.detid()) {
-    return one.energy() < other.energy();
+namespace io_v1 {
+  inline bool operator<(const FTLRecHit& one, const FTLRecHit& other) {
+    if (one.detid() == other.detid()) {
+      return one.energy() < other.energy();
+    }
+    return one.detid() < other.detid();
   }
-  return one.detid() < other.detid();
-}
 
-inline bool operator<(const FTLRecHit& one, const uint32_t& detid) { return one.detid() < detid; }
+  inline bool operator<(const FTLRecHit& one, const uint32_t& detid) { return one.detid() < detid; }
 
-inline bool operator<(const uint32_t& detid, const FTLRecHit& other) { return detid < other.detid(); }
-
+  inline bool operator<(const uint32_t& detid, const FTLRecHit& other) { return detid < other.detid(); }
+}  // namespace io_v1
 #endif

--- a/DataFormats/FTLRecHit/interface/FTLUncalibratedRecHit.h
+++ b/DataFormats/FTLRecHit/interface/FTLUncalibratedRecHit.h
@@ -4,75 +4,77 @@
 #include <vector>
 #include "DataFormats/DetId/interface/DetId.h"
 
-class FTLUncalibratedRecHit {
-public:
-  typedef DetId key_type;
+namespace io_v1 {
+  class FTLUncalibratedRecHit {
+  public:
+    typedef DetId key_type;
 
-  enum Flags {
-    kGood = -1,  // channel is good (mutually exclusive with other states)  setFlagBit(kGood) reset flags_ to zero
-    kPoorReco,   // channel has been badly reconstructed (e.g. bad shape, bad chi2 etc.)
-    kSaturated,  // saturated channel
-    kOutOfTime   // channel out of time
+    enum Flags {
+      kGood = -1,  // channel is good (mutually exclusive with other states)  setFlagBit(kGood) reset flags_ to zero
+      kPoorReco,   // channel has been badly reconstructed (e.g. bad shape, bad chi2 etc.)
+      kSaturated,  // saturated channel
+      kOutOfTime   // channel out of time
+    };
+
+    FTLUncalibratedRecHit();
+    FTLUncalibratedRecHit(const DetId& detId,
+                          uint8_t row,
+                          uint8_t column,
+                          std::pair<float, float> ampl,
+                          std::pair<float, float> time,
+                          float timeError,
+                          float position,
+                          float positionError,
+                          unsigned char flags = 0);
+    FTLUncalibratedRecHit(const DetId& detId,
+                          std::pair<float, float> ampl,
+                          std::pair<float, float> time,
+                          float timeError,
+                          float position,
+                          float positionError,
+                          unsigned char flags = 0);
+
+    ~FTLUncalibratedRecHit() = default;
+    std::pair<float, float> amplitude() const { return amplitude_; }
+    std::pair<float, float> time() const { return time_; }
+    float position() const { return position_; }
+
+    float timeError() const { return timeError_; }
+    float positionError() const { return positionError_; }
+
+    unsigned char flags() const { return flags_; };
+
+    DetId id() const { return id_; }
+    int row() const { return row_; }
+    int column() const { return column_; }
+
+    void setAmplitude(std::pair<float, float> amplitude) { amplitude_ = amplitude; }
+    void setTime(std::pair<float, float> time) { time_ = time; }
+
+    void setTimeError(float timeErr) { timeError_ = timeErr; }
+    void setId(DetId id) { id_ = id; }
+    void setFlagBit(Flags flag);
+    bool checkFlag(Flags flag) const;
+
+    void setPosition(float position) { position_ = position; }
+    void setPositionError(float positionErr) { positionError_ = positionErr; }
+
+    bool isTimeValid() const;
+    bool isTimeErrorValid() const;
+
+    bool isSaturated() const;
+
+  private:
+    std::pair<float, float> amplitude_;
+    std::pair<float, float> time_;
+    float timeError_;
+    float position_;  /// distance from the center of the bar to the hit
+    float positionError_;
+    DetId id_;
+    uint8_t row_;
+    uint8_t column_;
+    unsigned char flags_;
   };
-
-  FTLUncalibratedRecHit();
-  FTLUncalibratedRecHit(const DetId& detId,
-                        uint8_t row,
-                        uint8_t column,
-                        std::pair<float, float> ampl,
-                        std::pair<float, float> time,
-                        float timeError,
-                        float position,
-                        float positionError,
-                        unsigned char flags = 0);
-  FTLUncalibratedRecHit(const DetId& detId,
-                        std::pair<float, float> ampl,
-                        std::pair<float, float> time,
-                        float timeError,
-                        float position,
-                        float positionError,
-                        unsigned char flags = 0);
-
-  ~FTLUncalibratedRecHit() = default;
-  std::pair<float, float> amplitude() const { return amplitude_; }
-  std::pair<float, float> time() const { return time_; }
-  float position() const { return position_; }
-
-  float timeError() const { return timeError_; }
-  float positionError() const { return positionError_; }
-
-  unsigned char flags() const { return flags_; };
-
-  DetId id() const { return id_; }
-  int row() const { return row_; }
-  int column() const { return column_; }
-
-  void setAmplitude(std::pair<float, float> amplitude) { amplitude_ = amplitude; }
-  void setTime(std::pair<float, float> time) { time_ = time; }
-
-  void setTimeError(float timeErr) { timeError_ = timeErr; }
-  void setId(DetId id) { id_ = id; }
-  void setFlagBit(Flags flag);
-  bool checkFlag(Flags flag) const;
-
-  void setPosition(float position) { position_ = position; }
-  void setPositionError(float positionErr) { positionError_ = positionErr; }
-
-  bool isTimeValid() const;
-  bool isTimeErrorValid() const;
-
-  bool isSaturated() const;
-
-private:
-  std::pair<float, float> amplitude_;
-  std::pair<float, float> time_;
-  float timeError_;
-  float position_;  /// distance from the center of the bar to the hit
-  float positionError_;
-  DetId id_;
-  uint8_t row_;
-  uint8_t column_;
-  unsigned char flags_;
-};
-
+}  // namespace io_v1
+using FTLUncalibratedRecHit = io_v1::FTLUncalibratedRecHit;
 #endif

--- a/DataFormats/FTLRecHit/src/FTLRecHit.cc
+++ b/DataFormats/FTLRecHit/src/FTLRecHit.cc
@@ -6,62 +6,63 @@
 namespace {
   constexpr float timereso_max = 10000;
 }
+namespace io_v1 {
+  FTLRecHit::FTLRecHit()
+      : FTLRecHit(DetId(), 0, 0, -1.f, -1.f, -1.f, -1.f, -1.f, std::numeric_limits<unsigned char>::max()) {}
 
-FTLRecHit::FTLRecHit()
-    : FTLRecHit(DetId(), 0, 0, -1.f, -1.f, -1.f, -1.f, -1.f, std::numeric_limits<unsigned char>::max()) {}
+  FTLRecHit::FTLRecHit(const DetId& id,
+                       uint8_t row,
+                       uint8_t column,
+                       float energy,
+                       float time,
+                       float timeError,
+                       float position,
+                       float positionError,
+                       uint32_t flagBits)
+      : id_(id),
+        energy_(energy),
+        time_(time),
+        timeError_(timeError),
+        position_(position),
+        positionError_(positionError),
+        row_(row),
+        column_(column),
+        flagBits_(flagBits) {}
 
-FTLRecHit::FTLRecHit(const DetId& id,
-                     uint8_t row,
-                     uint8_t column,
-                     float energy,
-                     float time,
-                     float timeError,
-                     float position,
-                     float positionError,
-                     uint32_t flagBits)
-    : id_(id),
-      energy_(energy),
-      time_(time),
-      timeError_(timeError),
-      position_(position),
-      positionError_(positionError),
-      row_(row),
-      column_(column),
-      flagBits_(flagBits) {}
+  FTLRecHit::FTLRecHit(
+      const DetId& id, float energy, float time, float timeError, float position, float positionError, uint32_t flagBits)
+      : FTLRecHit(id, 0, 0, energy, time, timeError, position, positionError, flagBits) {}
 
-FTLRecHit::FTLRecHit(
-    const DetId& id, float energy, float time, float timeError, float position, float positionError, uint32_t flagBits)
-    : FTLRecHit(id, 0, 0, energy, time, timeError, position, positionError, flagBits) {}
-
-bool FTLRecHit::isTimeValid() const {
-  if (timeError() < 0)
-    return false;
-  else
-    return true;
-}
-
-bool FTLRecHit::isTimeErrorValid() const {
-  if (!isTimeValid())
-    return false;
-  if (timeError() >= timereso_max)
-    return false;
-
-  return true;
-}
-
-/// check if one of the flags in a set is true
-bool FTLRecHit::checkFlags(const std::vector<int>& flagsvec) const {
-  for (std::vector<int>::const_iterator flagPtr = flagsvec.begin(); flagPtr != flagsvec.end();
-       ++flagPtr) {  // check if one of the flags is up
-    if (checkFlag(*flagPtr))
+  bool FTLRecHit::isTimeValid() const {
+    if (timeError() < 0)
+      return false;
+    else
       return true;
   }
-  return false;
-}
 
-std::ostream& operator<<(std::ostream& s, const FTLRecHit& hit) {
-  if (hit.detid().det() == DetId::Forward && hit.detid().subdetId() == FastTime)
-    return s << MTDDetId(hit.detid()) << ": " << hit.energy() << " GeV, " << hit.time() << " ns";
-  else
-    return s << "FTLRecHit undefined subdetector";
-}
+  bool FTLRecHit::isTimeErrorValid() const {
+    if (!isTimeValid())
+      return false;
+    if (timeError() >= timereso_max)
+      return false;
+
+    return true;
+  }
+
+  /// check if one of the flags in a set is true
+  bool FTLRecHit::checkFlags(const std::vector<int>& flagsvec) const {
+    for (std::vector<int>::const_iterator flagPtr = flagsvec.begin(); flagPtr != flagsvec.end();
+         ++flagPtr) {  // check if one of the flags is up
+      if (checkFlag(*flagPtr))
+        return true;
+    }
+    return false;
+  }
+
+  std::ostream& operator<<(std::ostream& s, const FTLRecHit& hit) {
+    if (hit.detid().det() == DetId::Forward && hit.detid().subdetId() == FastTime)
+      return s << MTDDetId(hit.detid()) << ": " << hit.energy() << " GeV, " << hit.time() << " ns";
+    else
+      return s << "FTLRecHit undefined subdetector";
+  }
+}  // namespace io_v1

--- a/DataFormats/FTLRecHit/src/classes_def.xml
+++ b/DataFormats/FTLRecHit/src/classes_def.xml
@@ -1,58 +1,53 @@
 <lcgdict>
-  <class name="FTLUncalibratedRecHit" ClassVersion="5">
-    <version ClassVersion="5" checksum="170707344"/>
-    <version ClassVersion="4" checksum="508372968"/>
-    <version ClassVersion="3" checksum="1325548734"/>
+  <class name="io_v1::FTLUncalibratedRecHit" ClassVersion="3">
+    <version ClassVersion="3" checksum="1613983504"/>
   </class>
-  <class name="std::vector<FTLUncalibratedRecHit>"/>
+  <class name="std::vector<io_v1::FTLUncalibratedRecHit>"/>
   <class name="FTLUncalibratedRecHitCollection"/>
   <class name="edm::Wrapper<FTLUncalibratedRecHitCollection>"/>
   <class name="edm::Ref<FTLUncalibratedRecHitCollection>"/>
   <class name="edm::RefVector<FTLUncalibratedRecHitCollection>"/>
   <class name="edm::RefProd<FTLUncalibratedRecHitCollection>"/>
 
-  <class name="FTLRecHit" ClassVersion="5">
-    <version ClassVersion="5" checksum="943006844"/>
-    <version ClassVersion="4" checksum="2916210238"/>
-    <version ClassVersion="3" checksum="259405474"/>
+  <class name="io_v1::FTLRecHit" ClassVersion="3">
+    <version ClassVersion="3" checksum="662557180"/>
   </class>
-  <class name="std::vector<FTLRecHit>"/>
+  <class name="std::vector<io_v1::FTLRecHit>"/>
   <class name="FTLRecHitCollection"/>
   <class name="edm::Wrapper<FTLRecHitCollection>" />
   <class name="edm::Ref<FTLRecHitCollection>"/>
   <class name="edm::RefVector<FTLRecHitCollection>"/>
   <class name="edm::RefProd<FTLRecHitCollection>"/>
 
-  <class name="FTLCluster" ClassVersion="4">
-   <version ClassVersion="4" checksum="291411646"/>
-    <version ClassVersion="3" checksum="2080981492"/>
+  <class name="io_v1::FTLCluster" ClassVersion="3">
+    <version ClassVersion="3" checksum="2743481150"/>
   </class>
-  <class name="std::vector<FTLCluster>"/>
+  <class name="std::vector<io_v1::FTLCluster>"/>
   <class name="FTLClusterCollection"/>
   <class name="edm::Wrapper<FTLClusterCollection>" />
-  <class name="edm::Ref<FTLClusterCollection,FTLCluster>"/>
+  <class name="edm::Ref<FTLClusterCollection,io_v1::FTLCluster>"/>
   <class name="edm::RefVector<FTLClusterCollection>"/>
   <class name="edm::RefProd<FTLClusterCollection>"/>
  
-  <class name="edm::DetSet<FTLRecHit>"/>
-  <class name="edm::DetSetVector<FTLRecHit>"/>
-  <class name="std::vector<edm::DetSet<FTLRecHit> >"/>
-  <class name="std::vector<std::vector<edm::DetSet<FTLRecHit> > >"/>
-  <class name="edm::Wrapper<edm::DetSet<FTLRecHit> >"/>
-  <class name="edm::Wrapper<std::vector<edm::DetSet<FTLRecHit> > >"/>
-  <class name="edm::Wrapper<edm::DetSetVector<FTLRecHit> >"/>
-  <class name="edm::Wrapper<std::vector<std::vector<edm::DetSet<FTLRecHit> > > >"/>
+  <class name="edm::DetSet<io_v1::FTLRecHit>"/>
+  <class name="edm::DetSetVector<io_v1::FTLRecHit>"/>
+  <class name="std::vector<edm::DetSet<io_v1::FTLRecHit> >"/>
+  <class name="std::vector<std::vector<edm::DetSet<io_v1::FTLRecHit> > >"/>
+  <class name="edm::Wrapper<edm::DetSet<io_v1::FTLRecHit> >"/>
+  <class name="edm::Wrapper<std::vector<edm::DetSet<io_v1::FTLRecHit> > >"/>
+  <class name="edm::Wrapper<edm::DetSetVector<io_v1::FTLRecHit> >"/>
+  <class name="edm::Wrapper<std::vector<std::vector<edm::DetSet<io_v1::FTLRecHit> > > >"/>
   
-  <class name="edmNew::DetSetVector<FTLCluster>"/>
-  <class name="edm::Wrapper<edmNew::DetSetVector<FTLCluster> >" />
-  <class name="edm::DetSetRefVector<FTLCluster,edm::DetSetVector<FTLCluster> >" />
+  <class name="edmNew::DetSetVector<io_v1::FTLCluster>"/>
+  <class name="edm::Wrapper<edmNew::DetSetVector<io_v1::FTLCluster> >" />
+  <class name="edm::DetSetRefVector<io_v1::FTLCluster,edm::DetSetVector<io_v1::FTLCluster> >" />
 
-  <class name="std::vector<edm::Ref<edmNew::DetSetVector<FTLCluster>,FTLCluster,edmNew::DetSetVector<FTLCluster>::FindForDetSetVector> >" />
-  <class name="edmNew::DetSetVector<edm::Ref<edmNew::DetSetVector<FTLCluster>,FTLCluster,edmNew::DetSetVector<FTLCluster>::FindForDetSetVector> >" />
-  <class name="edm::Wrapper<edmNew::DetSetVector<edm::Ref<edmNew::DetSetVector<FTLCluster>,FTLCluster,edmNew::DetSetVector<FTLCluster>::FindForDetSetVector> > >" />
+  <class name="std::vector<edm::Ref<edmNew::DetSetVector<io_v1::FTLCluster>,io_v1::FTLCluster,edmNew::DetSetVector<io_v1::FTLCluster>::FindForDetSetVector> >" />
+  <class name="edmNew::DetSetVector<edm::Ref<edmNew::DetSetVector<io_v1::FTLCluster>,io_v1::FTLCluster,edmNew::DetSetVector<io_v1::FTLCluster>::FindForDetSetVector> >" />
+  <class name="edm::Wrapper<edmNew::DetSetVector<edm::Ref<edmNew::DetSetVector<io_v1::FTLCluster>,io_v1::FTLCluster,edmNew::DetSetVector<io_v1::FTLCluster>::FindForDetSetVector> > >" />
   
   <class name="FTLTrackingRecHitFromHit" ClassVersion="3">
-    <version ClassVersion="3" checksum="3399436891"/>
+    <version ClassVersion="3" checksum="3143543127"/>
   </class>
   <class name="FTLTrackingRecHitCollection"/>
   <class name="edm::Wrapper<FTLTrackingRecHitCollection>"/>

--- a/SimDataFormats/Associations/src/classes_def.xml
+++ b/SimDataFormats/Associations/src/classes_def.xml
@@ -261,13 +261,13 @@
   <class name="std::pair<FTLClusterRef, std::vector<MtdSimLayerClusterRef> >" />
   <class name="std::vector<std::pair<FTLClusterRef, std::vector<MtdSimLayerClusterRef> > >" />
   <class name="MtdRecoClusterToSimLayerClusterAssociationMap" ClassVersion="3">
-    <version ClassVersion="3" checksum="3711078147"/>
+    <version ClassVersion="3" checksum="1411787573"/>
   </class>
   <class name="edm::Wrapper<MtdRecoClusterToSimLayerClusterAssociationMap>" />
   <class name="std::pair<MtdSimLayerClusterRef, std::vector<FTLClusterRef> >" />
   <class name="std::vector<std::pair<MtdSimLayerClusterRef, std::vector<FTLClusterRef> > >" />
   <class name="MtdSimLayerClusterToRecoClusterAssociationMap" ClassVersion="3">
-    <version ClassVersion="3" checksum="4259123457"/>
+    <version ClassVersion="3" checksum="3331727647"/>
   </class><class name="edm::Wrapper<MtdSimLayerClusterToRecoClusterAssociationMap>" />
   <class name="reco::MtdRecoClusterToSimLayerClusterAssociator" persistent="false"/>
   <class name="edm::Wrapper<reco::MtdRecoClusterToSimLayerClusterAssociator>" persistent="false"/>

